Development/nmos/client_utils.cpp

```diff
@@ -55,39 +60,165 @@ namespace nmos
+                    }
+#else
+                    slog::log<slog::severities::error>(gate, SLOG_FLF) << "Bind HTTP client connection to " << interface_name << " not supported";
+#endif
+                }
+            };
+        }
+
+#ifdef CPPRESTSDK_ENABLE_BIND_WEBSOCKET_CLIENT
+        // The current version of the C++ REST SDK 2.10.18 does not provide the callback to enable the custom websocket setting
+        inline std::function<void(web::websockets::client::native_handle)> make_ws_client_nativehandle_options(const utility::string_t& client_address, bool secure, slog::base_gate& gate)
+        {
+            // get the associated network interface name from IP address
+            const auto interface_name = get_interface_name(client_address);
+            if (!client_address.empty() && interface_name.empty())
+            {
+                slog::log<slog::severities::error>(gate, SLOG_FLF) << "No network interface found for " << client_address << " to bind for the websocket client connection";
+            }
+
+            return [interface_name, secure, &gate](web::websockets::client::native_handle native_handle)
+            {
+                if (!interface_name.empty())
+                {
+#if defined(linux) || defined(__linux) || defined(__linux__)
+                    if (!bind_to_device(interface_name, secure, native_handle))
+                    {
+                        char error[1024];
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Unable to bind websocket client connection to " << interface_name << ", bind_to_device reported error : " << strerror_r(errno, error, sizeof(error));
+                    }
+#else
+                    slog::log<slog::severities::error>(gate, SLOG_FLF) << "Bind websocket client connection to " << interface_name << " not supported";
+#endif
+                }
+            };
+        }
+#endif
+
 #endif
     }
 
-    // construct client config based on settings, e.g. using the specified proxy
+    // construct client config based on settings and specified secure flag, e.g. using the specified proxy, and the OCSP client
     // with the remaining options defaulted, e.g. request timeout
-    web::http::client::http_client_config make_http_client_config(const nmos::settings& settings, load_ca_certificates_handler load_ca_certificates, slog::base_gate& gate)
+    web::http::client::http_client_config make_http_client_config(const nmos::settings& settings, load_ca_certificates_handler load_ca_certificates, bool secure, slog::base_gate& gate)
     {
         web::http::client::http_client_config config;
         const auto proxy = proxy_uri(settings);
         if (!proxy.is_empty()) config.set_proxy(proxy);
-        config.set_validate_certificates(nmos::experimental::fields::validate_certificates(settings));
+        if (secure) config.set_validate_certificates(nmos::experimental::fields::validate_certificates(settings));
```

Comment:
Clearly previously we thought it didn't matter to do `set_validate_certificiates` and `set_ssl_context_callback` even for HTTP, they just wouldn't be used. What changed?

Reply:
As we have the `secure` variable, I thought it will look better not to call the `set_validate_certificates` if not in secure mode.

Reply:
We discussed that an OCSP client is insecure even when the client_secure setting is true.

We noted that `load_ca_certificates` would only be required when `secure` is false, so that feels a bit odd too.

Reply:
We discussed that `secure` is currently needed because of the type-erased `native_handle` types, so added a comment about that, and agreed to leave as is.